tests/cypress/tests/unit/clipboard_test.ts

```diff
@@ -3,13 +3,75 @@ import * as R from 'ramda';
 import applyClipboardToData from 'dash-table/utils/applyClipboardToData';
 
 describe('clipboard', () => {
+    const columns = ['c1'].map(id => ({ id: id, name: id, editable: true, sort_as_null: [] }));
+
+    describe('with hidden columns', () => {
+        const altColumns = [
+            'c1',
+            'c2'
+        ].map(id => ({ id: id, name: id, editable: true, sort_as_null: [] }));
+
+        describe('with column overflow', () => {
+            it('returns all columns, 1st column visible only', () => {
+                const altVisibleColumns = altColumns.slice(0, 1);
+
+                const res = applyClipboardToData(
+                    R.range(0, 2).map(value => [`${value}`, `${value + 1}`]),
+                    { row: 0, column: 0, column_id: '' },
+                    R.range(0, 1),
+                    altColumns,
+                    altVisibleColumns,
+                    R.range(0, 1).map(() => ({ c1: 'c1', c2: 'c2' })),
+                    true,
+                    true
+                );
+
+                expect(res).to.not.equal(undefined);
+
+                if (res) {
+                    expect(res.data.length).to.equal(2);
+                    expect(Object.entries(res.data[0]).length).to.equal(3);
+                    expect(res.columns.length).to.equal(3);
+                    expect(res.columns[0].id).to.equal('c1');
+                    expect(res.columns[2].id).to.equal('c2');
+                }
+            });
+        });
+
+        it('returns all columns, 2nd column visible only', () => {
+            const altVisibleColumns = altColumns.slice(-1);
+
+            const res = applyClipboardToData(
+                R.range(0, 2).map(value => [`${value}`, `${value + 1}`]),
+                { row: 0, column: 0, column_id: '' },
+                R.range(0, 1),
+                altColumns,
+                altVisibleColumns,
+                R.range(0, 1).map(() => ({ c1: 'c1', c2: 'c2' })),
+                true,
+                true
+            );
+
+            expect(res).to.not.equal(undefined);
+
+            if (res) {
+                expect(res.data.length).to.equal(2);
+                expect(Object.entries(res.data[0]).length).to.equal(3);
+                expect(res.columns.length).to.equal(3);
+                expect(res.columns[0].id).to.equal('c1');
+                expect(res.columns[1].id).to.equal('c2');
```

Comment:
Check that all columns are present and in the right order after a copy/paste operation involving hidden columns